src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
```

Comment:
this will require us to call GetAscii twice. may be this can be ok but did we think we can allocate extra memory instead of calling GetAscii twice? the ACE prefix is 4 characters so we can allocate like 2*count. note that most of the cases we encounter Ascii characters.